File: notification_service_rb/dependency-injection/file_handler.rb
require 'json'

class FileHandler
  def initialize(file_path)
    @file_path = file_path
  end

  def read_file
    return [] unless File.exist?(@file_path)

    file_content = File.read(@file_path)
    JSON.parse(file_content, symbolize_names: true)
  end

  def write_file(data)
    File.write(@file_path, JSON.pretty_generate(data))
  end
end

File: notification_service_js/dependency-injection-factory/FileHandler.js
const fs = require('fs');

class FileHandler {
  constructor(filePath) {
    this.filePath = filePath;
  }

  readFile() {
    if (fs.existsSync(this.filePath)) {
      const data = fs.readFileSync(this.filePath, 'utf-8');
      return JSON.parse(data);
    }
    return [];
  }

  writeFile(data) {
    fs.writeFileSync(this.filePath, JSON.stringify(data, null, 2));
  }
}

module.exports = FileHandler;
